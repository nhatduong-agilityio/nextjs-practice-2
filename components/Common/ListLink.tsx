// Libs
import React, { Fragment, memo } from 'react'
import { List as ChakraList, ListItem as ChakraListItem, Text } from '@chakra-ui/react'
import NextLink from 'next/link'

type TListLink = {
  id: string
  name: string
}

interface IProps {
  items: TListLink[]
}

const ListLink = memo(({ items }: IProps) => (
  <ChakraList spacing={['10px', '15px']}>
    {items.map((list) => (
      <Fragment key={list.id}>
        <ChakraListItem>
          <NextLink href='/' passHref>
            <Text data-testid='link' size={['14', '16']}>
              {list.name}
            </Text>
          </NextLink>
        </ChakraListItem>
      </Fragment>
    ))}
  </ChakraList>
))

ListLink.displayName = 'ListLink'

export default ListLink
