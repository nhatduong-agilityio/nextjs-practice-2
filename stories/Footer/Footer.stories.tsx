// Components
import Footer from '@components/Footer/Footer'
import type { Meta } from '@storybook/react'

const meta: Meta<typeof Footer> = {
  title: 'Footer/Footer',
  component: Footer,
}

export default meta

export const Default = {
  render: () => <Footer />,
}
