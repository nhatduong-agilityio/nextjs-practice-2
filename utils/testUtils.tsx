import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
