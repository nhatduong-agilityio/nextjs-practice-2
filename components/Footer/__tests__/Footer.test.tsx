import { render, screen, waitFor } from '@utils/testUtils'
import React from 'react'
import ListLink from '@components/Common/ListLink'
import { AboutUsLinks, CommunityLinks, ContactUsLinks, HelpAndSupportLinks } from '@mock/dataMock'
import Footer from '../Footer'

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}))

describe('Footer', () => {
  it('Should Footer match Snapshot', () => {
    const { container } = render(<Footer />)

    expect(container).toMatchSnapshot()
  })

  it('Should renders the community links', async () => {
    render(<Footer />)

    const communityLink = screen.getByText('Community')

    expect(communityLink).toBeInTheDocument()

    const { container } = render(<ListLink items={CommunityLinks} />)

    let lazyContainer

    await waitFor(() => {
      lazyContainer = container
    })

    expect(lazyContainer).toMatchSnapshot()
  })

  it('Should renders the About Us links', async () => {
    render(<Footer />)

    const aboutUsLink = screen.getByText('About Us')

    expect(aboutUsLink).toBeInTheDocument()

    const { container } = render(<ListLink items={AboutUsLinks} />)

    let lazyContainer

    await waitFor(() => {
      lazyContainer = container
    })

    expect(lazyContainer).toMatchSnapshot()
  })

  it('Should renders the Help & Support links', async () => {
    render(<Footer />)

    const supportLink = screen.getByText('Help & Support')

    expect(supportLink).toBeInTheDocument()

    const { container } = render(<ListLink items={HelpAndSupportLinks} />)

    let lazyContainer

    await waitFor(() => {
      lazyContainer = container
    })

    expect(lazyContainer).toMatchSnapshot()
  })

  it('Should renders the Contact Us links', async () => {
    render(<Footer />)

    const contactLink = screen.getByText('Contact Us')

    expect(contactLink).toBeInTheDocument()

    const { container } = render(<ListLink items={ContactUsLinks} />)

    let lazyContainer

    await waitFor(() => {
      lazyContainer = container
    })

    expect(lazyContainer).toMatchSnapshot()
  })

  it('Should renders other links', async () => {
    render(<Footer />)

    const termOfService = screen.getAllByText('Term of Service')
    const sitemap = screen.getAllByText('Sitemap')
    const privacyPolicy = screen.getAllByText('Privacy Policy')
    const partners = screen.getAllByText('Partners & Affiliates')
    const copyRight = screen.getAllByText('2019 Â© Test - All right reserved')

    expect(termOfService[0]).toBeInTheDocument()
    expect(sitemap[0]).toBeInTheDocument()
    expect(privacyPolicy[0]).toBeInTheDocument()
    expect(partners[0]).toBeInTheDocument()
    expect(copyRight[0]).toBeInTheDocument()
  })
})
