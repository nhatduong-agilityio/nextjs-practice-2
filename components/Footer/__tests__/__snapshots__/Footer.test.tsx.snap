// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Should Footer match Snapshot 1`] = `
<div>
  <footer
    class="chakra-container css-15vovlo"
  >
    <div
      class="css-1wuihf8"
    >
      <div
        class="chakra-stack css-227doo"
      >
        <div
          class="chakra-stack css-8g8ihq"
        >
          <a
            aria-label="brand-store"
            href="/"
          >
            <div
              class="chakra-stack css-uk2lwn"
            >
              <svg
                class="chakra-icon css-2k9v9y"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22.7178 7.68579C22.7096 7.57546 22.6172 7.49011 22.5067 7.49011H18.6948V6.43125C18.6948 3.0448 15.9396 0.289856 12.5534 0.289856C9.16715 0.289856 6.412 3.0448 6.412 6.43125V7.49011H2.6001C2.48955 7.49011 2.39722 7.57546 2.38896 7.68579L0.906552 27.1688C0.902105 27.2275 0.922223 27.2855 0.96246 27.3287C1.00248 27.3721 1.05882 27.3967 1.11769 27.3967H23.9891C24.048 27.3967 24.1043 27.3721 24.1443 27.3289C24.1844 27.2857 24.2047 27.2277 24.2002 27.169L22.7178 7.68579ZM8.10617 6.43125C8.10617 3.97914 10.1011 1.98403 12.5534 1.98403C15.0057 1.98403 17.0006 3.97914 17.0006 6.43125V7.49011H8.10617V6.43125ZM7.25909 13.4197C6.09116 13.4197 5.14136 12.4697 5.14136 11.302C5.14136 10.4568 5.56491 9.6953 6.412 9.36112V11.302C6.412 11.7692 6.79213 12.1491 7.25909 12.1491C7.72604 12.1491 8.10617 11.7692 8.10617 11.302V9.36239C8.95326 9.69657 9.37681 10.4638 9.37681 11.302C9.37681 12.4697 8.42701 13.4197 7.25909 13.4197ZM17.8477 13.4197C16.6798 13.4197 15.73 12.4697 15.73 11.302C15.73 10.4568 16.1535 9.6953 17.0006 9.36112V11.302C17.0006 11.7692 17.3807 12.1491 17.8477 12.1491C18.3147 12.1491 18.6948 11.7692 18.6948 11.302V9.36239C19.5419 9.69657 19.9654 10.4638 19.9654 11.302C19.9654 12.4697 19.0156 13.4197 17.8477 13.4197Z"
                />
              </svg>
              <svg
                class="chakra-icon css-uemw57"
                focusable="false"
                viewBox="0 0 63 15"
              >
                <path
                  d="M7.03922 8.17836C7.03922 8.77838 6.88701 9.31222 6.58259 9.77989C6.27817 10.2431 5.83918 10.6049 5.26563 10.8652C4.69208 11.1211 4.01926 11.2491 3.24718 11.2491C2.60304 11.2491 2.06258 11.2049 1.6258 11.1167C1.18902 11.024 0.734588 10.8652 0.262512 10.6402V8.31072C0.761059 8.56661 1.27946 8.76735 1.81771 8.91294C2.35597 9.05413 2.8501 9.12472 3.30012 9.12472C3.68837 9.12472 3.97294 9.05854 4.15383 8.92618C4.33472 8.78941 4.42516 8.61514 4.42516 8.40337C4.42516 8.27101 4.38766 8.1563 4.31266 8.05924C4.24207 7.95776 4.12515 7.85629 3.96191 7.75482C3.80308 7.65334 3.37512 7.44598 2.67804 7.13273C2.04713 6.84596 1.57285 6.56801 1.25519 6.29888C0.941948 6.02975 0.708116 5.72092 0.553699 5.37238C0.403694 5.02384 0.328691 4.61132 0.328691 4.13483C0.328691 3.24363 0.652967 2.54875 1.30152 2.0502C1.95007 1.55165 2.84128 1.30238 3.97514 1.30238C4.97665 1.30238 5.99801 1.53401 7.03922 1.99726L6.23846 4.01571C5.33402 3.60099 4.55311 3.39363 3.89573 3.39363C3.55601 3.39363 3.30894 3.45319 3.15453 3.57231C3.00011 3.69144 2.9229 3.83924 2.9229 4.01571C2.9229 4.20542 3.01996 4.37528 3.21409 4.52529C3.41262 4.67529 3.94647 4.94883 4.81562 5.34591C5.64947 5.72092 6.22743 6.12461 6.5495 6.55698C6.87598 6.98494 7.03922 7.5254 7.03922 8.17836Z"
                />
                <path
                  d="M13.1012 11.1167V7.02685C13.1012 6.02093 12.8034 5.51797 12.2078 5.51797C11.7842 5.51797 11.471 5.69665 11.2681 6.05402C11.0695 6.40697 10.9702 6.99597 10.9702 7.82099V11.1167H8.38266V0.819275H10.9702V2.28183C10.9702 2.94362 10.935 3.69805 10.8644 4.54514H10.9835C11.2306 4.15689 11.5239 3.88335 11.8637 3.72453C12.2034 3.5657 12.5982 3.48628 13.0483 3.48628C13.8865 3.48628 14.5373 3.72894 15.0005 4.21425C15.4682 4.69515 15.702 5.37017 15.702 6.23932V11.1167H13.1012Z"
                />
                <path
                  d="M24.6229 7.35112C24.6229 8.57764 24.292 9.53503 23.6303 10.2233C22.9729 10.9071 22.0508 11.2491 20.864 11.2491C19.7257 11.2491 18.8191 10.8983 18.144 10.1968C17.4734 9.49532 17.1381 8.54676 17.1381 7.35112C17.1381 6.12902 17.4668 5.18046 18.1242 4.50544C18.786 3.826 19.7125 3.48628 20.9037 3.48628C21.6405 3.48628 22.2912 3.6429 22.856 3.95615C23.4207 4.2694 23.8553 4.71941 24.1597 5.3062C24.4685 5.88857 24.6229 6.57021 24.6229 7.35112ZM19.7654 7.35112C19.7654 7.99527 19.8514 8.4872 20.0235 8.82691C20.1956 9.16222 20.4846 9.32987 20.8905 9.32987C21.2919 9.32987 21.5743 9.16222 21.7375 8.82691C21.9052 8.4872 21.989 7.99527 21.989 7.35112C21.989 6.7114 21.9052 6.22829 21.7375 5.90181C21.5699 5.57533 21.2831 5.41209 20.8772 5.41209C20.4801 5.41209 20.1956 5.57533 20.0235 5.90181C19.8514 6.22388 19.7654 6.70698 19.7654 7.35112Z"
                />
                <path
                  d="M30.6055 11.2491C30.2261 11.2491 29.8864 11.1873 29.5864 11.0638C29.2863 10.9358 28.9885 10.7086 28.6929 10.3821H28.5738C28.6532 10.9071 28.6929 11.2336 28.6929 11.3616V14.3727H26.1053V3.61864H28.2098L28.5738 4.57161H28.6929C29.1738 3.84806 29.8378 3.48628 30.6849 3.48628C31.5144 3.48628 32.1651 3.83041 32.6372 4.51867C33.1093 5.20693 33.3453 6.15108 33.3453 7.35112C33.3453 8.55999 33.0982 9.51297 32.6041 10.2101C32.1144 10.9027 31.4482 11.2491 30.6055 11.2491ZM29.7386 5.51797C29.3591 5.51797 29.0922 5.65474 28.9378 5.92828C28.7878 6.19741 28.7062 6.59889 28.6929 7.13273V7.33789C28.6929 7.98203 28.779 8.44749 28.951 8.73426C29.1231 9.02104 29.3944 9.16442 29.765 9.16442C30.0959 9.16442 30.3364 9.02324 30.4864 8.74088C30.6364 8.45411 30.7114 7.98203 30.7114 7.32465C30.7114 6.68051 30.6364 6.21947 30.4864 5.94152C30.3364 5.65915 30.0871 5.51797 29.7386 5.51797Z"
                />
                <path
                  d="M38.5006 1.44136H41.8758C43.1905 1.44136 44.1656 1.63769 44.8009 2.03035C45.4406 2.4186 45.7605 3.02524 45.7605 3.85027C45.7605 4.39293 45.6215 4.85618 45.3435 5.24002C45.0656 5.62386 44.6994 5.87093 44.245 5.98122V6.0474C44.845 6.20623 45.2796 6.47095 45.5487 6.84155C45.8178 7.21215 45.9524 7.69305 45.9524 8.28425C45.9524 9.16222 45.6215 9.85489 44.9597 10.3623C44.3023 10.8652 43.4045 11.1167 42.2662 11.1167H38.5006V1.44136ZM41.1147 5.15399H41.9022C42.2772 5.15399 42.5662 5.07678 42.7692 4.92236C42.9765 4.76794 43.0802 4.53852 43.0802 4.2341C43.0802 3.69144 42.6699 3.4201 41.8493 3.4201H41.1147V5.15399ZM41.1147 7.06656V9.09825H42.0346C42.8508 9.09825 43.2589 8.75412 43.2589 8.06586C43.2589 7.74379 43.1486 7.49672 42.928 7.32465C42.7118 7.15259 42.3964 7.06656 41.9816 7.06656H41.1147Z"
                />
                <path
                  d="M52.3717 11.1167L51.8754 10.124H51.8224C51.4739 10.5564 51.1187 10.852 50.757 11.0108C50.3952 11.1696 49.9275 11.2491 49.354 11.2491C48.6481 11.2491 48.0922 11.0373 47.6863 10.6137C47.2804 10.1902 47.0774 9.59459 47.0774 8.82691C47.0774 8.02836 47.3554 7.43495 47.9113 7.0467C48.4672 6.65404 49.2724 6.43345 50.3268 6.38491L51.5776 6.34521V6.23932C51.5776 5.62165 51.2732 5.31282 50.6643 5.31282C50.1172 5.31282 49.4224 5.49812 48.5797 5.86872L47.8319 4.16131C48.7054 3.71129 49.8084 3.48628 51.1408 3.48628C52.1026 3.48628 52.846 3.72453 53.371 4.20101C53.9005 4.6775 54.1652 5.3437 54.1652 6.19961V11.1167H52.3717ZM50.4525 9.42252C50.7658 9.42252 51.0327 9.32325 51.2533 9.12472C51.4783 8.92618 51.5908 8.66808 51.5908 8.35043V7.76805L50.9952 7.79452C50.1437 7.82541 49.718 8.13865 49.718 8.73426C49.718 9.1931 49.9628 9.42252 50.4525 9.42252Z"
                />
                <path
                  d="M63 3.61864V4.86942L61.9743 5.24664C62.1155 5.50253 62.186 5.80033 62.186 6.14005C62.186 6.94743 61.9015 7.57393 61.3323 8.01953C60.7676 8.46514 59.9227 8.68794 58.7977 8.68794C58.5197 8.68794 58.2991 8.67029 58.1359 8.63499C58.0741 8.7497 58.0433 8.8578 58.0433 8.95927C58.0433 9.08722 58.1469 9.18648 58.3543 9.25708C58.5616 9.32325 58.833 9.35634 59.1683 9.35634H60.4125C62.0934 9.35634 62.9339 10.0645 62.9339 11.4807C62.9339 12.394 62.5434 13.1043 61.7625 13.6116C60.986 14.119 59.8941 14.3727 58.4866 14.3727C57.4234 14.3727 56.6028 14.194 56.0248 13.8366C55.4512 13.4837 55.1645 12.9741 55.1645 12.3079C55.1645 11.4123 55.7204 10.8343 56.8322 10.574C56.6028 10.477 56.402 10.3203 56.2299 10.1042C56.0623 9.88798 55.9785 9.674 55.9785 9.46223C55.9785 9.2284 56.0424 9.02324 56.1704 8.84677C56.2983 8.66588 56.5653 8.4519 56.9712 8.20483C56.5829 8.03277 56.2763 7.76364 56.0513 7.39745C55.8307 7.03126 55.7204 6.58566 55.7204 6.06064C55.7204 5.24443 55.9983 4.61132 56.5542 4.16131C57.1101 3.71129 57.9043 3.48628 58.9367 3.48628C59.0734 3.48628 59.3095 3.50172 59.6448 3.53261C59.9801 3.56349 60.2271 3.59217 60.386 3.61864H63ZM57.3484 12.0498C57.3484 12.2616 57.4631 12.4315 57.6925 12.5594C57.9219 12.6918 58.2285 12.7579 58.6124 12.7579C59.2389 12.7579 59.7396 12.6785 60.1146 12.5197C60.4941 12.3653 60.6838 12.1645 60.6838 11.9175C60.6838 11.7189 60.5691 11.5777 60.3397 11.4939C60.1102 11.4145 59.7815 11.3748 59.3536 11.3748H58.3411C58.0631 11.3748 57.8271 11.4388 57.6329 11.5667C57.4432 11.6991 57.3484 11.8601 57.3484 12.0498ZM58.1624 6.07387C58.1624 6.84155 58.4293 7.22539 58.9631 7.22539C59.2102 7.22539 59.3999 7.13053 59.5323 6.94082C59.669 6.7511 59.7374 6.46653 59.7374 6.08711C59.7374 5.31061 59.4793 4.92236 58.9631 4.92236C58.4293 4.92236 58.1624 5.3062 58.1624 6.07387Z"
                />
              </svg>
            </div>
          </a>
          <p
            class="chakra-text css-1u87w9r"
          >
            We are the bed furniture platform. We are already working on thousands of future home projects. Trust us, you will surely be satisfied
          </p>
        </div>
        <div
          class="css-1m1rgq5"
          style="margin: 0px;"
        >
          <div
            class="chakra-stack css-4a3jos"
          >
            <p
              class="chakra-text css-liv8y5"
            >
              Community
            </p>
            <ul
              class="css-3cgtfu"
              role="list"
            >
              <li
                class="css-0"
              >
                <a
                  href="/"
                >
                  <p
                    class="chakra-text css-63f6z0"
                    data-testid="link"
                  >
                    Job Board
                  </p>
                </a>
              </li>
              <li
                class="css-0"
              >
                <a
                  href="/"
                >
                  <p
                    class="chakra-text css-63f6z0"
                    data-testid="link"
                  >
                    Career Advice
                  </p>
                </a>
              </li>
              <li
                class="css-0"
              >
                <a
                  href="/"
                >
                  <p
                    class="chakra-text css-63f6z0"
                    data-testid="link"
                  >
                    Pricing
                  </p>
                </a>
              </li>
              <li
                class="css-0"
              >
                <a
                  href="/"
                >
                  <p
                    class="chakra-text css-63f6z0"
                    data-testid="link"
                  >
                    Assessments
                  </p>
                </a>
              </li>
              <li
                class="css-0"
              >
                <a
                  href="/"
                >
                  <p
                    class="chakra-text css-63f6z0"
                    data-testid="link"
                  >
                    Powered
                  </p>
                </a>
              </li>
            </ul>
          </div>
          <div
            class="chakra-stack css-4a3jos"
          >
            <p
              class="chakra-text css-liv8y5"
            >
              About Us
            </p>
            <ul
              class="css-3cgtfu"
              role="list"
            >
              <li
                class="css-0"
              >
                <a
                  href="/"
                >
                  <p
                    class="chakra-text css-63f6z0"
                    data-testid="link"
                  >
                    Career
                  </p>
                </a>
              </li>
              <li
                class="css-0"
              >
                <a
                  href="/"
                >
                  <p
                    class="chakra-text css-63f6z0"
                    data-testid="link"
                  >
                    Internship
                  </p>
                </a>
              </li>
              <li
                class="css-0"
              >
                <a
                  href="/"
                >
                  <p
                    class="chakra-text css-63f6z0"
                    data-testid="link"
                  >
                    Press
                  </p>
                </a>
              </li>
              <li
                class="css-0"
              >
                <a
                  href="/"
                >
                  <p
                    class="chakra-text css-63f6z0"
                    data-testid="link"
                  >
                    Blog
                  </p>
                </a>
              </li>
              <li
                class="css-0"
              >
                <a
                  href="/"
                >
                  <p
                    class="chakra-text css-63f6z0"
                    data-testid="link"
                  >
                    Contact
                  </p>
                </a>
              </li>
            </ul>
          </div>
          <div
            class="chakra-stack css-4a3jos"
          >
            <p
              class="chakra-text css-liv8y5"
            >
              Help & Support
            </p>
            <ul
              class="css-3cgtfu"
              role="list"
            >
              <li
                class="css-0"
              >
                <a
                  href="/"
                >
                  <p
                    class="chakra-text css-63f6z0"
                    data-testid="link"
                  >
                    Job Search
                  </p>
                </a>
              </li>
              <li
                class="css-0"
              >
                <a
                  href="/"
                >
                  <p
                    class="chakra-text css-63f6z0"
                    data-testid="link"
                  >
                    Hello
                  </p>
                </a>
              </li>
              <li
                class="css-0"
              >
                <a
                  href="/"
                >
                  <p
                    class="chakra-text css-63f6z0"
                    data-testid="link"
                  >
                    Hiring
                  </p>
                </a>
              </li>
              <li
                class="css-0"
              >
                <a
                  href="/"
                >
                  <p
                    class="chakra-text css-63f6z0"
                    data-testid="link"
                  >
                    Talent Guide
                  </p>
                </a>
              </li>
              <li
                class="css-0"
              >
                <a
                  href="/"
                >
                  <p
                    class="chakra-text css-63f6z0"
                    data-testid="link"
                  >
                    Online Test
                  </p>
                </a>
              </li>
            </ul>
          </div>
          <div
            class="chakra-stack css-4a3jos"
          >
            <div
              class="css-1fwdkaw"
            >
              <p
                class="chakra-text css-liv8y5"
              >
                Contact Us
              </p>
              <ul
                class="css-3cgtfu"
                role="list"
              >
                <li
                  class="css-0"
                >
                  <a
                    href="/"
                  >
                    <p
                      class="chakra-text css-63f6z0"
                      data-testid="link"
                    >
                      Hello@shopbag.com
                    </p>
                  </a>
                </li>
                <li
                  class="css-0"
                >
                  <a
                    href="/"
                  >
                    <p
                      class="chakra-text css-63f6z0"
                      data-testid="link"
                    >
                      081437650889
                    </p>
                  </a>
                </li>
              </ul>
              <div
                class="chakra-stack css-pa2ktp"
                style="margin: 0px;"
              >
                <a
                  aria-label="social-fb"
                  href="/"
                >
                  <svg
                    class="chakra-icon css-1yyzf6i"
                    focusable="false"
                    viewBox="0 0 20 20"
                  >
                    <path
                      d="M20 3.33356C20 1.582 18.4187 0 16.6667 0H3.33333C1.58133 0 0 1.582 0 3.33356V16.6664C0 18.418 1.58133 20 3.33356 20H10V12.4444H7.55556V9.11111H10V7.81244C10 5.57267 11.6818 3.55556 13.75 3.55556H16.4444V6.88889H13.75C13.4551 6.88889 13.1111 7.24689 13.1111 7.78311V9.11111H16.4444V12.4444H13.1111V20H16.6667C18.4187 20 20 18.418 20 16.6664V3.33356Z"
                    />
                  </svg>
                </a>
                <a
                  aria-label="social-tw"
                  href="/"
                >
                  <svg
                    class="chakra-icon css-1yyzf6i"
                    focusable="false"
                    viewBox="0 0 20 20"
                  >
                    <path
                      d="M18 0H2C0.9 0 0 0.9 0 2V18C0 19.1 0.9 20 2 20H18C19.1 20 20 19.1 20 18V2C20 0.9 19.1 0 18 0ZM15.7 7.3C15.6 11.9 12.7 15.1 8.3 15.3C6.5 15.4 5.2 14.8 4 14.1C5.3 14.3 7 13.8 7.9 13C6.6 12.9 5.8 12.2 5.4 11.1C5.8 11.2 6.2 11.1 6.5 11.1C5.3 10.7 4.5 10 4.4 8.4C4.7 8.6 5.1 8.7 5.5 8.7C4.6 8.2 4 6.3 4.7 5.1C6 6.5 7.6 7.7 10.2 7.9C9.5 5.1 13.3 3.6 14.8 5.5C15.5 5.4 16 5.1 16.5 4.9C16.3 5.6 15.9 6 15.4 6.4C15.9 6.3 16.4 6.2 16.8 6C16.7 6.5 16.2 6.9 15.7 7.3Z"
                    />
                  </svg>
                </a>
                <a
                  aria-label="social-li"
                  href="/"
                >
                  <svg
                    class="chakra-icon css-1yyzf6i"
                    focusable="false"
                    viewBox="0 0 20 20"
                  >
                    <path
                      d="M18.5231 0H1.47784C0.661648 0 0.000106812 0.64136 0.000106812 1.43296V18.5665C0.000106812 19.3581 0.66188 20 1.47784 20H18.5231C19.3391 20 19.9999 19.3577 19.9999 18.5665V1.43296C19.9999 0.641593 19.3391 0 18.5231 0ZM6.06348 16.7414H3.04165V7.71162H6.06348V16.7414ZM4.55287 6.47801H4.53255C3.5192 6.47801 2.86231 5.78498 2.86231 4.91736C2.86231 4.03225 3.53849 3.35802 4.57193 3.35802C5.60547 3.35802 6.2412 4.03207 6.26129 4.91736C6.26129 5.78517 5.6057 6.47801 4.55287 6.47801ZM16.9563 16.7414H13.9354V11.9102C13.9354 10.6962 13.4977 9.86787 12.4053 9.86787C11.5696 9.86787 11.0735 10.4267 10.8556 10.9662C10.7749 11.1589 10.7549 11.4284 10.7549 11.6976V16.7412H7.73461C7.73461 16.7412 7.77413 8.55837 7.73461 7.71148H10.7554V8.99165C11.1564 8.37651 11.873 7.49903 13.4771 7.49903C15.4654 7.49903 16.9563 8.78887 16.9563 11.5637V16.7414ZM10.7358 9.02034C10.741 9.01187 10.7482 9.00169 10.7554 8.99179V9.02034H10.7358Z"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-1rarrvt"
        style="margin: 0px;"
      >
        <div
          class="css-9sn4kq"
        >
          <a
            href="/"
          >
            <p
              class="chakra-text css-63f6z0"
            >
              Term of Service
            </p>
          </a>
          <a
            href="/"
          >
            <p
              class="chakra-text css-63f6z0"
            >
              Sitemap
            </p>
          </a>
          <a
            href="/"
          >
            <p
              class="chakra-text css-63f6z0"
            >
              Privacy Policy
            </p>
          </a>
          <a
            href="/"
          >
            <p
              class="chakra-text css-63f6z0"
            >
              Partners & Affiliates
            </p>
          </a>
        </div>
        <p
          class="chakra-text css-69kteu"
        >
          2019 © Test - All right reserved
        </p>
        <div
          class="css-civijs"
        >
          <a
            href="/"
          >
            <p
              class="chakra-text css-63f6z0"
            >
              Term of Service
            </p>
          </a>
          <a
            href="/"
          >
            <p
              class="chakra-text css-63f6z0"
            >
              Privacy Policy
            </p>
          </a>
          <a
            href="/"
          >
            <p
              class="chakra-text css-63f6z0"
            >
              Partners & Affiliates
            </p>
          </a>
          <a
            href="/"
          >
            <p
              class="chakra-text css-63f6z0"
            >
              Sitemap
            </p>
          </a>
        </div>
      </div>
    </div>
  </footer>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;

exports[`Footer Should renders the About Us links 1`] = `
<div>
  <ul
    class="css-3cgtfu"
    role="list"
  >
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Career
        </p>
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Internship
        </p>
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Press
        </p>
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Blog
        </p>
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Contact
        </p>
      </a>
    </li>
  </ul>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;

exports[`Footer Should renders the Contact Us links 1`] = `
<div>
  <ul
    class="css-3cgtfu"
    role="list"
  >
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Hello@shopbag.com
        </p>
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          081437650889
        </p>
      </a>
    </li>
  </ul>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;

exports[`Footer Should renders the Help & Support links 1`] = `
<div>
  <ul
    class="css-3cgtfu"
    role="list"
  >
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Job Search
        </p>
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Hello
        </p>
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Hiring
        </p>
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Talent Guide
        </p>
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Online Test
        </p>
      </a>
    </li>
  </ul>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;

exports[`Footer Should renders the community links 1`] = `
<div>
  <ul
    class="css-3cgtfu"
    role="list"
  >
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Job Board
        </p>
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Career Advice
        </p>
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Pricing
        </p>
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Assessments
        </p>
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-63f6z0"
          data-testid="link"
        >
          Powered
        </p>
      </a>
    </li>
  </ul>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
